compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2603,6 +2603,11 @@ private boolean validateActionParentNode(Location pos, BLangNode node) {
         return false;
     }
 
+    private boolean checkActionInQuery(NodeKind parentKind) {
+        return parentKind == NodeKind.FROM || parentKind == NodeKind.SELECT ||
```

Comment:
This should be allowed isn't it?
```ballerina
    _ = from var i in 1...3
        where ([] == from int ii in [1] let var res = start getInt(1) select i)
        select i;
```

Reply:
Equality expression only allows `expressions` as its subexpressions?
https://ballerina.io/spec/lang/master/#section_6.29

Reply:
for equality it's anyway an expression, but can the nested be an action?